src/components/IncompleteSubReqCourse.vue

```diff
@@ -137,6 +148,16 @@ export default Vue.extend({
       }
       this.courseObjects = firstFourCourseObjects;
     },
+    onShowAllCourses(courses: AppCourse[]) {
+      this.$emit('onShowAllCourses', {
+        requirementName: this.subReq.requirement.name,
+        subReqCoursesArray: this.subReqCoursesNotTakenArray,
+      });
+    },
+    onAddCourse() {
+      const dashboardRef = this.$parent.$parent.$parent.$parent as any;
```

Comment:
I remember peeping one of Theresa's PRs and seeing Sam comment "👀👀" about the `$parent.$parent.$parent.$parent` so here's me doing the same 

👀👀

Reply:
O LOOOL YES I wanted to pre-comment about this but I forgot

Reply:
Yes, I do acknowledge that this is an absolutely disgusting practice. I would not recommend doing this on any circumstances... BUT... I actually believe it makes the most sense in this situation because the add course model is so integrated with the way we structure the data up to the individual semester component. If we want to use emit, we would have to move the model component to Dashboard.vue and in doing so would migrate half of the code away from semester, and honestly make things a lot more confusing because we would also need to pass back all the courses data from each semester. 

If we were to rebuild CoursePlan, we would've designed entire components tree with this in mind so it wouldn't have to do it. Otherwise, I imagine the refactor would take at least two days, which at the current stage is not that well worth it compared to .$parent.

Reply:
you've convinced me